### PR TITLE
Docs/release notes 0.37.0

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
Cleaning up 0.37.0 release notes

**Implementation Description (bullets)**:
- made 0.37.0 table
- cleaned up wording for notes in current release
- Changes in this release for reference:

> a39c0c7	Glenn Sheasby	2017-03-29	Sweep Metrics Part I: Cells deleted and examined (#1695)
> eeeacfb	gmaretic	2017-03-28	Fix/ptsmrtse (#1718)
> 0b85913	Chris Lockfort	2017-03-28	fix for internal artifactory publish NPEing (#1744)
> 33467ac	Jeremy Kong	2017-03-27	TimeLock Rollback Documentation (#1716)
> 9e55d62	gmaretic	2017-03-27	Feature/ts benchmark (#1720)
> b5804df	Himangi Saraogi	2017-03-27	Merge pull request #1736 from palantir/fix/unset-loaded-config
> 09bffae	gmaretic	2017-03-24	Benchmark/fix tables (#1732)
> 24486ff	gmaretic	2017-03-23	[no release notes] (#1731)
> 680ffbd	Jeremy Kong	2017-03-22	Don't silent-retry RemoteLockService connections (#1727)
> 211d2ad	Chris Lockfort	2017-03-22	bump database dependencies and collapse metrics versions (#1524)
> 84e9f68	Chris Lockfort	2017-03-22	remove deprecated guavaisms (#1717)
> ff85398	gmaretic	2017-03-22	Fix/benchmarks (#1723)
> 8f51498	Chris Lockfort	2017-03-17	Merge pull request #1710 from palantir/deleteRangeBenchmarks
> 6f07528	Glenn Sheasby	2017-03-17	Merge pull request #1689 from palantir/remove_get_async_rows
> 4b46440	Robert Hero	2017-03-17	Merge pull request #1713 from palantir/docs/improve-dw-metrics
> 003b86a	Harry Biddle	2017-03-17	Merge pull request #1547 from palantir/add-log-parsing-script
> 5c150cf	Glenn Sheasby	2017-03-16	Merge pull request #1265 from palantir/feature/loadSingleStream
> efdef88	Jeremy Kong	2017-03-16	Document Learnings from Deployment to the Field (#1711)
> e7e230b	Glenn Sheasby	2017-03-16	Merge pull request #1702 from palantir/fix/oracle-colrange
> d2d006b	Robert Hero	2017-03-16	Merge pull request #1707 from palantir/docs/release-notes-0.36.0
> ed4d62e	John Boreiko	2017-03-16	Fix migration cli not updating the timestamp on ToKVS (#1199)
> 86ab8f8	Tom Petracca	2017-03-15	Merge pull request #1708 from palantir/fix_docs
> 1f66e6e	Robert Hero	2017-03-15	move #1684 and #1681 to the proper release notes section
> ac8ca6e	Robert Hero	2017-03-15	Merge pull request #1699 from palantir/docs/dropwizard-metrics
> 6f097c2	Greg Bonik	2017-03-14	Add a release note for #1702

**Concerns (what feedback would you like?)**:
Make sure everything makes sense.

**Where should we start reviewing?**:
release-docs.rts

**Priority (whenever / two weeks / yesterday)**:
Today

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -49,12 +55,8 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1720>`__)
 
     *    - |fixed|
-         - Fixed an issue where a ``MultipleRunningTimestampServicesError`` would be ignored, resulting in a state where two timestamp services would be able to simultaneously hand out timestamps. Also changed the logic for increasing the timestamp bound when the allocation buffer is exhausted.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/1718>`__)
-
-    *    - |fixed|
          - KVS migrations where timestamp data was co-located with AtlasDB data now respect the timestamp service contract.
-           Previously, doing a KVS migration with an embedded timestamp service whose timestamp data is co-located with the AtlasDB data causes timestamps to reset to the logical beginning of time.
+           Previously, doing a KVS migration with an embedded timestamp service whose timestamp data is co-located with the AtlasDB data was causing timestamps to reset to the logical beginning of time.
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: was causing -> caused.

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -67,9 +69,9 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1265>`__)
 
     *    - |devbreak|
-         - 'getAsyncRows' and 'getAsyncRowsMultimap' methods have been removed from generated code.  They do not appear valuable to the API and use a nonintuitive and custom 'AsyncProxy' (also removed).
+         - ``getAsyncRows`` and ``getAsyncRowsMultimap`` methods have been removed from generated code.  They do not appear valuable to the API and use an unintuitive and custom ``AsyncProxy`` that was also removed.
            We believe they are unused by upstream applications, but if you do encounter breaks due to this
-           removal please file a ticket with the dev team for immediate support (as we did not take the time to properly deprecate the methods).
+           removal please file a ticket with the dev team for immediate support since the methods were not properly deprecated.
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: nit: comma before "since"?

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -80,7 +82,9 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1727>`__)
 
     *    - |fixed| |improved|
-         - Cassandra depedencies have been bumped to newer versions; should fix a bug (#1654) that caused Atlas probing downed Cassandra nodes every few minutes to see if they were up and working yet to eventually take out the entire cluster by steadily building up leaked connections, due to a bug in the underlying driver.
+         - Cassandra depedencies have been bumped to newer versions; should fix a bug (`#1654 <https://github.com/palantir/atlasdb/issues/1654>`__) that caused
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: Would be nice to have a stress-test for this, but wouldn't block the release on it.

Comment (gmaretic) [inline docs/source/release_notes/release-notes.rst]: Good point. Opened an issue [#1753](https://github.com/palantir/atlasdb/issues/1753) to track.

File: docs/source/release_notes/release-notes.rst [hunk 4]
```diff
@@ -30,15 +30,21 @@ Changelog
...
+29 Mar 2017
+
 .. list-table::
     :widths: 5 40
     :header-rows: 1
 
     *    - Type
          - Change
+    *    - |fixed|
+         - Fixed an issue where a ``MultipleRunningTimestampServicesError`` would be ignored, resulting in a state where two timestamp services would be able to simultaneously hand out timestamps.
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: What does this mean for people who don't have this fix? Does this mean I could have silent data corruption? I'd try to communicate how likely this is to happen, or how serious of an issue this is.

File: docs/source/release_notes/release-notes.rst [hunk 5]
```diff
@@ -30,15 +30,21 @@ Changelog
...
+           The new behaviour is that, once the error is encountered, requesting a fresh timestamp throws. Also improved the logic for increasing the timestamp bound when the allocation buffer is exhausted.
+           (`Pull Request <https://github.com/palantir/atlasdb/pull/1718>`__)
 
     *    - |new|
          - Added :ref:`Dropwizard metrics <dropwizard-metrics>` for sweep, exposing aggregate and table-specific counts of cells examined and stale values deleted.
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: Thanks for linking to the docs!

File: docs/source/release_notes/release-notes.rst [hunk 6]
```diff
@@ -49,12 +55,8 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1720>`__)
 
     *    - |fixed|
-         - Fixed an issue where a ``MultipleRunningTimestampServicesError`` would be ignored, resulting in a state where two timestamp services would be able to simultaneously hand out timestamps. Also changed the logic for increasing the timestamp bound when the allocation buffer is exhausted.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/1718>`__)
-
-    *    - |fixed|
          - KVS migrations where timestamp data was co-located with AtlasDB data now respect the timestamp service contract.
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: The wording here is hard to parse. I would say this like 

"KVS migrations now maintain the guarantee of the timestamp service to hand out monotonically increasing timestamps. Previously, we would reset the timestamp service to 0 after a migration, but now we use the correct logical timestamp.

Note: this does not affect users who are using the Timelock server"

File: docs/source/release_notes/release-notes.rst [hunk 7]
```diff
@@ -67,9 +69,9 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1265>`__)
 
     *    - |devbreak|
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: I would remove the last sentence about not properly deprecating. If we are confident that this API isn't used, then let's say so. If we aren't sure, then let's revert the break, deprecate it, and remove it in a release or two.

My guess is we are very confident it's not used by anybody and it was just broken before, so there's little risk to removing it.

File: docs/source/release_notes/release-notes.rst [hunk 8]
```diff
@@ -67,9 +69,9 @@ develop
...
-         - 'getAsyncRows' and 'getAsyncRowsMultimap' methods have been removed from generated code.  They do not appear valuable to the API and use a nonintuitive and custom 'AsyncProxy' (also removed).
+         - ``getAsyncRows`` and ``getAsyncRowsMultimap`` methods have been removed from generated code.  They do not appear valuable to the API and use an unintuitive and custom ``AsyncProxy`` that was also removed.
            We believe they are unused by upstream applications, but if you do encounter breaks due to this
-           removal please file a ticket with the dev team for immediate support (as we did not take the time to properly deprecate the methods).
+           removal please file a ticket with the dev team for immediate support since the methods were not properly deprecated.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1689>`__)
 
     *    - |fixed|
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: We should highlight that this only affects products using the Timelock server with HTTP/2 configured (and link to docs).

File: docs/source/release_notes/release-notes.rst [hunk 9]
```diff
@@ -80,7 +82,9 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1727>`__)
 
     *    - |fixed| |improved|
-         - Cassandra depedencies have been bumped to newer versions; should fix a bug (#1654) that caused Atlas probing downed Cassandra nodes every few minutes to see if they were up and working yet to eventually take out the entire cluster by steadily building up leaked connections, due to a bug in the underlying driver.
+         - Cassandra depedencies have been bumped to newer versions; should fix a bug (`#1654 <https://github.com/palantir/atlasdb/issues/1654>`__) that caused
+           Atlas probing downed Cassandra nodes every few minutes to see if they were up and working yet to eventually take out the entire cluster by steadily
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: The description of the bug isn't the cleanest, but I don't think it's that big of a deal. We should highlight what the  version is that we actually bumped to, and we should just pick one of "fixed" or "improved" (I vote for "fixed").

File: docs/source/release_notes/release-notes.rst [hunk 10]
```diff
@@ -49,38 +76,37 @@ develop
...
 
     *    - |fixed|
-         - Fixed an issue where a ``MultipleRunningTimestampServicesError`` would be ignored, resulting in a state where two timestamp services would be able to simultaneously hand out timestamps. Also changed the logic for increasing the timestamp bound when the allocation buffer is exhausted.
-           (`Pull Request <https://github.com/palantir/atlasdb/pull/1718>`__)
-
-    *    - |fixed|
-         - KVS migrations where timestamp data was co-located with AtlasDB data now respect the timestamp service contract.
-           Previously, doing a KVS migration with an embedded timestamp service whose timestamp data is co-located with the AtlasDB data causes timestamps to reset to the logical beginning of time.
+         - KVS migrations now maintain the guarantee of the timestamp service to hand out monotonically increasing timestamps.
+           Previously, we would reset the timestamp service to 0 after a migration, but now we use the correct logical timestamp.
```
Comment (hsaraogi) [inline docs/source/release_notes/release-notes.rst]: nit: `zero`?
